src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts

```diff
@@ -399,55 +452,81 @@ export class FeedPostIconRowComponent {
     comp.globalVars._alertError("Transaction broadcast successfully but read node timeout exceeded. Please refresh.");
   }
 
-  diamondPopoverOpen = false;
-  openDiamondPopover() {
-    this.backendApi.SetStorage("hasSeenDiamondInfo", true);
-    this.collapseDiamondInfo = this.backendApi.GetStorage("collapseDiamondInfo");
-    this.diamondSelected = this.getCurrentDiamondLevel();
-    this.diamondPopover.show();
-    this.diamondPopoverOpen = true;
-    document.addEventListener("click", this.popoverOpenClickHandler, true);
-  }
-
-  closeDiamondPopover() {
-    this.diamondPopoverOpen = false;
-    this.diamondPopover.hide();
-    document.removeEventListener("click", this.popoverOpenClickHandler);
-  }
-
   popoverOpenClickHandler = (e: Event) => {
     const popoverElement = document.getElementById("diamond-popover");
     if (popoverElement && e.target !== popoverElement && !popoverElement.contains(e.target as any)) {
       e.stopPropagation();
-      this.closeDiamondPopover();
     }
   };
-  expandDiamondInfo(event: any): void {
-    this.toggleDiamondInfo(event, false);
+
+  async sendOneDiamond(event: any, fromDragEvent: boolean) {
+    // Disable diamond selection if diamonds are being sent
+    if (this.sendingDiamonds) {
+      return;
+    }
+
+    // Block user from selecting diamond level below already gifted amount
+    if (this.getCurrentDiamondLevel() > 0) {
+      return;
+    }
+
+    // Don't trigger diamond purchases on tap on tablet
+    if (event && event.pointerType === "touch" && !fromDragEvent) {
```

Comment:
how do I send a single diamond on tablet / mobile?

Reply:
On tablet, you can tap the diamond to open the pop-up, and click again to select the amount. On mobile you tap to send a single diamond, or drag to select another amount.